packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -749,33 +785,46 @@ function parseShape(configuration: Configuration, path: string, value: unknown,
     if (!subDefinition)
       throw new UsageError(`Unrecognized configuration settings found: ${path}.${propKey} - run "yarn config -v" to see the list of settings supported in Yarn`);
 
-    result.set(propKey, parseValue(configuration, subPath, propValue, definition.properties[propKey], folder));
+    const newValue = parseValue(configuration, subPath, propValue, definition.properties[propKey], folder, {overwrite, reset});
+
+    if (newValue !== undefined) {
+      result.set(propKey, newValue);
+    }
   }
 
-  return result;
+  return new Map(overwrite
+    ? [...currentValue, ...result]
+    : [...result, ...currentValue],
+  );
```

Comment:
Why isn't `overwrite` overwriting? 🤔

```suggestion
  return new Map(overwrite
    ? result
    : [...result, ...currentValue],
  );
```

Reply:
I intend to avoid mutating the currentValue, but after you asked, I found that my parseShape did not avoid 😵.

after onConflict was implemented, I did some refactoring of this, I think it should be more readable now.